Scripts/muemu.py

```diff
@@ -0,0 +1,267 @@
+#
+#  Script for running QEMU with the appropriate options for the given SKU/ARCH.
+#
+#  Copyright (c) Microsoft Corporation
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+
+import zipfile
+import urllib.request
+import sys
+import subprocess
+import shutil
+import requests
+import os
+import argparse
+from typing import List
+
+#
+# Constants
+#
+
+DEFAULT_VERSION = "4.7.3"
+
+#
+# Setup and parse arguments.
+#
+
+parser = argparse.ArgumentParser()
+
+# HINT: Run with '--help all' to get complete help.
+parser.add_argument("-u", "--update", action="store_true",
+                    help="Updates the firmware binaries.")
+parser.add_argument("--firmwaredir", default="./firmware",
+                    help="Directory to download and use firmware binaries.")
+parser.add_argument("-a", "--arch", default="x64",
+                    choices=["x64", "arm64"], help="The guest architecture for the VM.")
+parser.add_argument("-d", "--disk",
+                    help="Path to the disk file.")
+parser.add_argument("-c", "--cores", default=2, type=int,
+                    help="The number of cores for the VM. This may be overridden based on the configuration.")
+parser.add_argument("-m", "--memory", default="4096",
+                    help="The memory size to use in Mb.")
+parser.add_argument("--tpm", action="store_true",
+                    help="Linux only. Enables the emulated TPM using swtpm.")
+parser.add_argument("--vnc",
+                    help="Provides the VNC port to use based on 5900. E.g. ':1' for localhost:5901")
+parser.add_argument("--accel", default="tcg",
+                    choices=["tcg", "kvm", "whpx"], help="Acceleration back-end to use in QEMU.")
+parser.add_argument("--version", default=DEFAULT_VERSION,
+                    help="The Project MU firmware version to use.")
+parser.add_argument("--qemudir", default="",
+                    help="Path to a custom QEMU install directory.")
+parser.add_argument("--gdbport", type=int,
+                    help="Enables the GDB server on the specified port. E.g. 1234")
+parser.add_argument("--debugfw", action="store_true",
+                    help="Enables update to use the DEBUG firmware binaries.")
+parser.add_argument("--verbose", action="store_true",
+                    help="Enabled verbose script prints.")
+parser.add_argument("--force", action="store_true",
+                    help="Disables automatic correction of VM configurations.")
+parser.add_argument("--timeout", type=int, default=None,
+                    help="The number of seconds to wait before killing the QEMU process.")
+parser.add_argument("--qemuargs", type=str, default=None,
+                    help="Additional arguments to provide to QEMU, using the format \" -foo bar\". "
+                    "The preceding space is required for parsing reasons.")

```

Comment:
Looking for suggestions on this. Annoyingly argparse seems to really freak out when `-` is the first character of a string, and so i have to require a space at the beginning for it to read the whole string as one thing.

Reply:
I think if you use `parse_known_args`, the result will become "known_args, left_over = parser.parse_known_args()"